administration/src/components/generation/GenerationController.tsx

```diff
@@ -30,17 +31,30 @@ const GenerationController = () => {
     }
 
     const confirm = async () => {
-        try {
-            setMode(Mode.loading)
-            const pdfDataUri = await generateCards(client, cardCreationModels, region)
-            downloadDataUri(pdfDataUri, "ehrenamtskarten.pdf")
-            setMode(Mode.finished)
-        } catch (e) {
-            console.error(e)
-            AppToaster.show({message: "Etwas ist schiefgegangen.", intent: "danger"})
-            setMode(Mode.input)
-        }
+        setMode(Mode.loading);
+        (await generateCards(client, cardCreationModels, region)).unwrap(
+            (blob: Blob) => {
+                downloadDataUri(blob, "ehrenamtskarten.pdf");
+                setMode(Mode.finished);
+            },
+            (error: PDFError) => {
+                console.error(error)
+                switch (error.type) {
+                    case "pdf-generation":
+                        AppToaster.show({message: "Etwas ist schiefgegangen.", intent: "danger"})
+                        break;
+                    case "unicode":
+                        AppToaster.show({message: `Ein Zeichen konnte nicht in der PDF eingebunden werden: ${error.unsupportedChar}`, intent: "danger"})
+                        break;
```

Comment:
Is the `switch` statement exhaustive in typescript?

Reply:
kind of, I think there is at least a warning